clients/java/zts/core/src/main/java/com/yahoo/athenz/zts/ZTSClient.java

```diff
@@ -413,7 +414,21 @@ public ZTSClient(String ztsUrl, Principal identity) {
     public ZTSClient(String ztsUrl, SSLContext sslContext) {
         this(ztsUrl, null, sslContext);
     }
-    
+
+    /**
+     * Constructs a new ZTSClient object with the given SSLContext object
+     * and ZTS Server Url. Default read and connect timeout values are
+     * 30000ms (30sec). The application can change these values by using the
+     * athenz.zts.client.read_timeout and athenz.zts.client.connect_timeout
+     * system properties. The values specified for timeouts must be in milliseconds.
+     * @param ztsUrl ZTS Server's URL (optional)
+     * @param sslContext SSLContext that includes service's private key and x.509 certificate
+     * @param notificationSender If provided, will be used for sending notifications on failures
+     */
+    public ZTSClient(String ztsUrl, SSLContext sslContext, ZTSClientNotificationSender notificationSender) {
```

Comment:
The reasons I put it in the constructor are the following:
1. I wanted the PrefetchTask thread to have access to the notificationSender. 
2. I didn't want the notificationSender to be static as I didn't want every instance of ZTSClient to have access to it (as more than one ZTSClient might be used by the same program and some might need to send notifications while others don't).

Because the PrefetchTask is an internal static class (and isn't aware of the outer class instance variables), I had to either pass them along through the constructor before the PrefetchTask starts or set it as a static member of ZTSClient. If it is acceptable I'll set it as a static member of ZTSClient and make it thread-safe (locking when setting it and when sending notifications).

Reply:
Where do we start the PrefetchTask within the ZTSClient constructor that requires the notification sender to be set as part of the same constructor?

I'm not following why would we miss the notification sender if we just have:

ZTSClient ztsClient = new ZTSClient(...);
ztsClient.setNotificationSender(notificationSender);
ztsClient.getAWSTemporaryCredentials(...);

Reply:
Got it. I accidentally looked at AWSCredentialsProviderImpl and ZTSClient as a single unit (the AWSCredentialsProviderImpl creates the ZTSClient and then get the credentials at the CTOR).
Fixing it...